### PR TITLE
Add org.qgis.qgis

### PR DESCRIPTION
**Pull request by aleixpol**

### CODE REVIEW

File: org.qgis.qgis.json [hunk 1]
```diff
@@ -0,0 +1,379 @@
+
+{
+    "id": "org.qgis.qgis",
+    "branch": "5.11",
+    "runtime": "org.kde.Platform",
+    "runtime-version": "5.11",
+    "sdk": "org.kde.Sdk",
+    "command": "qgis",
+    "tags": ["nightly"],
```
Comment (TingPing) [inline org.qgis.qgis.json]: Remove tag

File: org.qgis.qgis.json [hunk 2]
```diff
@@ -0,0 +1,379 @@
+
+{
+    "id": "org.qgis.qgis",
+    "branch": "5.11",
```
Comment (TingPing) [inline org.qgis.qgis.json]: Branch is `stable`

File: org.qgis.qgis.json [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+        {
+            "name": "qgis",
+            "buildsystem": "cmake-ninja",
+            "sources": [
+            {
+                "type": "git", "url": "https://github.com/qgis/QGIS.git", "branch": "final-3_2_1"
+            },
+            {
+                "type": "patch",
+                "---source": "https://issues.qgis.org/attachments/download/13041/qgis-3.2.1-fix-sip-name.patch",
```
Comment (TingPing) [inline org.qgis.qgis.json]: Comment lines should start with `//`

File: org.qgis.qgis.json [hunk 4]
```diff
@@ -0,0 +1,379 @@
...
+                "-DQWT_LIBRARY=/app/lib/libqwt.so"
+            ],
+            "modules": [
+            {
+                "name": "proj",
+                "sources": [ { "type": "archive", "url": "https://github.com/OSGeo/proj.4/releases/download/5.1.0/proj-5.1.0.tar.gz", "sha256": "6b1379a53317d9b5b8c723c1dc7bf2e3a8eb22ceb46b8807a1ce48ef65685bb3" } ]
+            },
+            {
+                "name": "geos",
+                "sources": [ { "type": "archive", "url": "http://download.osgeo.org/geos/geos-3.6.0.tar.bz2", "sha256": "1fe7644f3240c8422b0143830ff142e44e8b01333c12f67681ccaab92142f2ea" }
```
Comment (TingPing) [inline org.qgis.qgis.json]: Please put every property on a new line.

File: org.qgis.qgis.json [hunk 5]
```diff
@@ -0,0 +1,379 @@
...
+                        "type": "archive",
+                        "url": "http://ftp.halifax.rwth-aachen.de/gnu/gsl/gsl-2.3.tar.gz",
+                        "sha256": "562500b789cd599b3a4f88547a7a3280538ab2ff4939504c8b4ac4ca25feadfb"
+                    }
+                ]
+            },
+            {
+                "name": "pyqt5",
+                "config-opts": [ "--disable-static", "--enable-x11" ],
+                "no-autogen": true,
```
Comment (TingPing) [inline org.qgis.qgis.json]: Unused property.

File: org.qgis.qgis.json [hunk 6]
```diff
@@ -0,0 +1,379 @@
...
+                "buildsystem": "simple",
+                "build-commands": [
+                    "python3 configure.py --assume-shared --confirm-license --no-designer-plugin --no-qml-plugin --sysroot=/app --verbose --qsci-api --qsci-api-destdir=/app/qsci --sipdir=/app/share/sip --sip=/app/bin/sip --sip-incdir=/app/include QMAKE_CFLAGS_RELEASE='-I/usr/include/python3.5m/' QMAKE_CXXFLAGS_RELEASE='-I/usr/include/python3.5m/'",
+                    "make -j `nproc`",
```
Comment (TingPing) [inline org.qgis.qgis.json]: Replace `nproc` with `$FLATPAK_BUILDER_N_JOBS`.

**Comment by TingPing** [comment]
bot, build org.qgis.qgis

File: org.qgis.qgis.json [hunk 7]
```diff
@@ -0,0 +1,459 @@
...
+                    "sources" : [
+                        {
+                            "type" : "archive",
+                            "url" : "http://www.gaia-gis.it/gaia-sins/libspatialite-sources/libspatialite-4.3.0a.tar.gz",
+                            "sha256" : "88900030a4762904a7880273f292e5e8ca6b15b7c6c3fb88ffa9e67ee8a5a499"
+                        }
+                    ],
+                    "modules" : [
+                        {
+                            "name" : "freexl",
```
Comment (nedrichards) [inline org.qgis.qgis.json]: This doesn't build on aarch64. It needs it's configure regenerated https://flathub.org/builds/#/builders/39/builds/301 an example of doing this is https://github.com/flathub/shared-modules/blob/08e9b9eead7a95139be5d1fbc24bea730fb78f7a/SDL/SDL_net-1.2.8.json

File: org.qgis.qgis.json [hunk 8]
```diff
@@ -0,0 +1,459 @@
...
+    ],
+    "separate-locales" : false,
+    "build-options" : {
+        "env" : {
+            "PYTHONPATH" : "/app/lib/python3.5/site-packages:/usr/lib/python3.5/site-packages"
+        }
+    },
+    "modules" : [
+        {
+            "name" : "qt5-qtwebkit",
```
Comment (eszlari) [inline org.qgis.qgis.json]: You could use https://github.com/flathub/io.qt.qtwebkit.BaseApp like e.g. https://github.com/flathub/org.musescore.MuseScore does, but it is based on 5.9 at the moment.

File: org.qgis.qgis.json [hunk 9]
```diff
@@ -0,0 +1,459 @@
...
+            "buildsystem" : "cmake-ninja",
+            "sources" : [
+                {
+                    "type" : "git",
+                    "url" : "https://github.com/qgis/QGIS.git",
+                    "branch" : "final-3_2_1"
+                },
+                {
+                    "type" : "patch",
+                    "//source" : "https://issues.qgis.org/attachments/download/13041/qgis-3.2.1-fix-sip-name.patch",
```
Comment (eszlari) [inline org.qgis.qgis.json]: If you used YAML instead of JSON, you could have proper comments.

File: org.qgis.qgis.json [hunk 10]
```diff
@@ -0,0 +1,459 @@
...
+            ],
+            "cleanup-platform" : [
+                "/bin",
+                "/mkspecs"
+            ],
+            "sources" : [
+                {
+                    "type" : "git",
+                    "url" : "git://code.qt.io/qt/qtwebkit.git",
+                    "branch" : "5.212"
```
Comment (eszlari) [inline org.qgis.qgis.json]: I think you need to choose a commit to make it reproducible.

File: org.qgis.qgis.json [hunk 11]
```diff
@@ -0,0 +1,459 @@
...
+                        }
+                    ],
+                    "build-commands" : [
+                        "sed -e 's|\\(target.path\\) = .*|\\1 = /app/lib|' -i Qt4Qt5/qscintilla.pro",
+                        "sed -e 's|\\(header.path\\) = .*|\\1 = /app/include|' -i Qt4Qt5/qscintilla.pro",
+                        "sed -e 's|\\(trans.path\\) = .*|\\1 = /app/share/qt5/translations|' -i Qt4Qt5/qscintilla.pro",
+                        "sed -e 's|\\(qsci.path\\) = .*|\\1 = /app/share/qt5|' -i Qt4Qt5/qscintilla.pro",
+                        "sed -e 's|\\(features.path\\) = .*|\\1 = /app/lib/qt5/mkspecs/features|' -i Qt4Qt5/qscintilla.pro",
+                        "sed -e \"s|qmake = {'CONFIG': 'qscintilla2'}|qmake = {'CONFIG': 'qscintilla2', 'QT': 'widgets printsupport'}|\" -i Python/configure.py",
+                        "cd Qt4Qt5 && qmake && make -j `nproc`",
```
Comment (eszlari) [inline org.qgis.qgis.json]: $FLATPAK_BUILDER_N_JOBS

File: org.qgis.qgis.json [hunk 12]
```diff
@@ -0,0 +1,459 @@
...
+            "buildsystem" : "simple",
+            "sources" : [
+                {
+                    "type" : "archive",
+                    "url" : "https://downloads.sourceforge.net/project/numpy/NumPy/1.11.2/numpy-1.11.2.tar.gz",
+                    "sha256" : "04db2fbd64e2e7c68e740b14402b25af51418fc43a59d9e54172b38b906b0f69"
+                }
+            ],
+            "build-commands" : [
+                "python3 setup.py build -j 4",
```
Comment (eszlari) [inline org.qgis.qgis.json]: $FLATPAK_BUILDER_N_JOBS

File: org.qgis.qgis.json [hunk 13]
```diff
@@ -0,0 +1,459 @@
...
+            ],
+            "sources" : [
+                {
+                    "type" : "archive",
+                    "url" : "https://pypi.python.org/packages/4d/de/32d741db316d8fdb7680822dd37001ef7a448255de9699ab4bfcbdf4172b/MarkupSafe-1.0.tar.gz",
+                    "sha256" : "a6be69091dac236ea9c6bc7d012beab42010fa914c459791d627dad4910eb665"
+                }
+            ],
+            "build-commands" : [
+                "python3 setup.py build -j `nproc`",
```
Comment (eszlari) [inline org.qgis.qgis.json]: $FLATPAK_BUILDER_N_JOBS

File: org.qgis.qgis.json (hunk 8 above)
Comment (aleixpol) [reply org.qgis.qgis.json]: I was explicitly asked by qgis developers to use Qt 5.11.

**Comment by aleixpol** [comment]
bot, build org.qgis.qgis

**Comment by nedrichards** [comment]
bot, build org.qgis.qgis

**Comment by eszlari** [comment]
@aleixpol 

`io.qt.qtwebkit.BaseApp` has a `5.11` branch now!

**Comment by aleixpol** [comment]
Is that really the reason why this has been stuck for about a month?

**Comment by barthalion** [comment]
No, it stalled because of unresolved comment about using branch instead of a commit for qtwebkit. With base app for 5.11 it can be solved even better now.

**Comment by nedrichards** [comment]
bot, build org.qgis.qgis

**Comment by nedrichards** [comment]
@aleixpol this is also still blocked by https://github.com/flathub/flathub/pull/529#pullrequestreview-143406448

File: org.qgis.qgis.json (hunk 7 above)
Comment (aleixpol) [reply org.qgis.qgis.json]: Isn't it what's being done few lines below?

Comment (nedrichards) [reply org.qgis.qgis.json]: Not for spatialite, that I can see in the manifest https://flathub.org/builds/#/builders/37/builds/358/steps/5/logs/stdio

Comment (aleixpol) [reply org.qgis.qgis.json]: ah, should be fixed for spatialite now

**Comment by nedrichards** [comment]
bot, build org.qgis.qgis

**Comment by nedrichards** [comment]
Numpy failed on arm https://flathub.org/builds/#/builders/47/builds/351 but aside from that, yes, fine. The only thing that seems obvious there is replacing:

> "python3 setup.py build -j 4",

with

> "python3 setup.py build -j $FLATPAK_BUILDER_N_JOBS",

as per the review comment

**Comment by aleixpol** [comment]
Replaced it, I don't really see the error message it had though.

**Comment by nedrichards** [comment]
bot, build org.qgis.qgis

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/org.qgis.qgis

**Comment by nedrichards** [comment]
@aleixpol anyone else we should add to the repo?

**Comment by aleixpol** [comment]
I'll see who's the right people to add. Can I do that or should I ask you?

**Comment by nedrichards** [comment]
@aleixpol I think you can do that as an admin. If you have any problems feel free to ask me.

**Comment by heis2201** [comment]
Wow, that's exciting... we will see Qgis as flatpak. Many thanks for the extra effort!

**Comment by aleixpol** [comment]
@TingPing are you sure that $FLATPAK_BUILDER_N_JOBS does something? On my flatpak-builder it's not defined so all jobs go wild with naked -j. :(

**Comment by eszlari** [comment]
What's you flatpak-builder version? FLATPAK_BUILDER_N_JOBS was introduced in 0.10.10!

**Comment by aleixpol** [comment]
Right, I'm running flatpak-builder 0.10.9 (on an ubuntu system, I guess it's the one coming with the distro)

**Pull request closed without merging**